### ISSUE TITLE
What size should my own dataset be set to?

### ISSUE THREAD

**Issue by zhujilin1995** [lightly-ai/lightly#1383]
Hello, your code is great. I have some questions to ask.

I have a binary classification task that I want to use the contrastive learning method to do. So I have the following questions: ① How should I prepare my own dataset? ②How many images should be prepared for each category? ③How many images should be prepared for the pretrain unlabeled data and the labeled data of the finetune part?

Thank you very much.

**Comment by zhujilin1995** [comment]
I have a binary classification task that I want to use the contrastive learning method to do. So I have the following questions: ① How should I prepare my data set? ②How many images should be prepared for each category? ③How many images should be prepared for the pretrain unlabeled data and the labeled data of the finetune part?

**Comment by guarin** [comment]
Hi, thanks for your interest in lightly! Those are some good, and difficult to answer, questions. I'll try my best :)

For 1) I am not sure I fully understand the question. But in general you would want to prepare the dataset as for a normal classification task (create train/val/test split etc.). The only difference is that you'll have some unlabeled training data. Depending on your domain you might also want to take class balance into account. It is generally better to have a more or less balanced dataset, also for contrastive pretraining. This is of course only an option if you have full control on how the dataset is created.

For 2) I would aim for at least 10'000 images for pretraining. Depending on your data and model it might also work with less data. But in general you want to use as much data as possible for pretraining. Regarding number of images per category I don't have any specific numbers, but I would try to avoid a scenario where you have many categories but only few images per category. For example, it will probably easier to get a good pretraining if you have 10 categories with 1'000 images each than if you have 1'000 categories with 10 images each.

3) really depends on your data. If it is a hard classification problem you'll need more labeled images than if the problem is simple. I would suggest to start with few labeled images, maybe 5-10% of the dataset. Then run pretraining on the unlabeled and labeled images and then finetune using only the labeled images. If the finetuning performance is too low you might want to label some more images and repeat the finetuning. Note that you can use all unlabeled and labeled train images for pretraining, so you only have to do it once. After that you just take unlabeled train images and label them until the finetuning performance is good enough.

Hope that helps :)

**Comment by zhujilin1995** [comment]
Thank you, this helps  a lot.
